client/dashboard/components/settings/general/store-address.js

```diff
@@ -0,0 +1,114 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { decodeEntities } from '@wordpress/html-entities';
+import { SelectControl, TextControl } from 'newspack-components';
+import { useMemo } from 'react';
+
+/**
+ * Form validation.
+ *
+ * @param {Object} values Keyed values of all fields in the form.
+ * @return {Object} Key value of fields and error messages, { myField: 'This field is required' }
+ */
+export function validateStoreAddress( values ) {
+	const errors = {};
+
+	if ( ! values.addressLine1.length ) {
+		errors.addressLine1 = __( 'Please add an address', 'woocommerce-admin' );
+	}
+	if ( ! values.countryState.length ) {
+		errors.countryState = __( 'Please select a country and state', 'woocommerce-admin' );
+	}
+	if ( ! values.city.length ) {
+		errors.city = __( 'Please add a city', 'woocommerce-admin' );
+	}
+	if ( ! values.postCode.length ) {
+		errors.postCode = __( 'Please add a post code', 'woocommerce-admin' );
```

Comment:
Probably outside the scope of this PR, but is it possible to further validate the value? "Banana" isn't a valid postcode anywhere 🍌

Reply:
I was able to use "banana 🍌"  for the actual WC settings.  I don't think any validation is applied here since it can vary so much between countries.

If we want to add in validation here, we might look into a package like [this one](https://github.com/melwynfurtado/postcode-validator).  But I'm also a little hesitant to introduce validation that does not (or may conflict with a store's settings) in WC.